### ISSUE TITLE
Installing v5.0.1-v5.0.3 throws warnings/errors for uninstalled peer dependencies 

### ISSUE THREAD

**Issue by MatthewCallis** [FormidableLabs/enzyme-matchers#217]
After upgrading to v5.0.1 I am getting several warning causes `npm ls` to fail. I am on React v16.2.0 and Jest v22.4.2.

After `npm install`:
```
npm WARN enzyme-adapter-react-13@1.0.3 requires a peer of react@^0.13.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-14@1.0.5 requires a peer of react@^0.14.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-14@1.0.5 requires a peer of react-addons-test-utils@^0.14.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-14@1.0.5 requires a peer of react-dom@^0.14.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15@1.0.5 requires a peer of react@^15.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15@1.0.5 requires a peer of react-dom@^15.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15@1.0.5 requires a peer of react-test-renderer@^15.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15.4@1.0.5 requires a peer of react@15.0.0-0 - 15.4.x but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15.4@1.0.5 requires a peer of react-dom@15.0.0-0 - 15.4.x but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15.4@1.0.5 requires a peer of react-addons-test-utils@15.0.0-0 - 15.4.x but none is installed. You must install peer dependencies yourself.
```

After `npm ls`:
```
├─┬ jest-enzyme@5.0.1
│ ├─┬ enzyme-matchers@5.0.1
│ │ ├── circular-json-es6@2.0.2
│ │ └─┬ deep-equal-ident@1.1.1
│ │   └─┬ lodash.isequal@3.0.4
│ │     ├─┬ lodash._baseisequal@3.0.7
│ │     │ ├── lodash.isarray@3.0.4 deduped
│ │     │ ├── lodash.istypedarray@3.0.6
│ │     │ └── lodash.keys@3.1.2 deduped
│ │     └── lodash._bindcallback@3.0.1
│ ├─┬ enzyme-to-json@3.3.1
│ │ └── lodash@4.17.5 deduped
│ └─┬ jest-environment-enzyme@5.0.1
│   ├── enzyme@3.3.0 deduped
│   ├─┬ enzyme-adapter-react-13@1.0.3
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-14@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15.4@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├── enzyme-adapter-react-16@1.1.1 deduped
│   ├── jest-environment-jsdom@22.4.1 deduped
│   ├── UNMET PEER DEPENDENCY react@^0.13.0
│   ├── UNMET PEER DEPENDENCY react-addons-test-utils@^0.14.0
│   ├── UNMET PEER DEPENDENCY react-dom@^0.14.0
│   └── UNMET PEER DEPENDENCY react-test-renderer@^15.5.0
...
npm ERR! peer dep missing: react@^0.13.0, required by enzyme-adapter-react-13@1.0.3
npm ERR! peer dep missing: react@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-dom@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-dom@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react-dom@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-test-renderer@^15.5.0, required by enzyme-adapter-react-15@1.0.5
```

**Comment by blainekasten** [comment]
ah. freaking lame :( 

I guess we'll have to make the environment have an unlisted dep on the adapters. and throw when not installed.

**Comment by MatthewCallis** [comment]
I've  updated to v5.0.2 and run into the same issue:

```
├─┬ jest-enzyme@5.0.2
│ ├─┬ enzyme-matchers@5.0.2
│ │ ├── circular-json-es6@2.0.2
│ │ └─┬ deep-equal-ident@1.1.1
│ │   └─┬ lodash.isequal@3.0.4
│ │     ├─┬ lodash._baseisequal@3.0.7
│ │     │ ├── lodash.isarray@3.0.4 deduped
│ │     │ ├── lodash.istypedarray@3.0.6
│ │     │ └── lodash.keys@3.1.2 deduped
│ │     └── lodash._bindcallback@3.0.1
│ ├─┬ enzyme-to-json@3.3.1
│ │ └── lodash@4.17.5 deduped
│ └─┬ jest-environment-enzyme@5.0.2
│   ├── enzyme@3.3.0 deduped
│   ├─┬ enzyme-adapter-react-13@1.0.3
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-14@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15.4@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├── enzyme-adapter-react-16@1.1.1 deduped
│   ├── jest-environment-jsdom@22.4.1 deduped
│   ├── UNMET PEER DEPENDENCY react@^0.13.0
│   ├── UNMET PEER DEPENDENCY react-addons-test-utils@^0.14.0
│   ├── UNMET PEER DEPENDENCY react-dom@^0.14.0
│   └── UNMET PEER DEPENDENCY react-test-renderer@^15.5.0
...
npm ERR! peer dep missing: react@^0.13.0, required by enzyme-adapter-react-13@1.0.3
npm ERR! peer dep missing: react@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-dom@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-dom@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react-dom@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-test-renderer@^15.5.0, required by enzyme-adapter-react-15@1.0.5
```

**Comment by ryan-roemer** [comment]
> I guess we'll have to make the environment have an unlisted dep on the adapters. and throw when not installed.

Probably the best move as they're separately named, can't really do a peerDeps semver thing. But, will conveniently avoid installing all the ones you're *not* using.

**Comment by fernandopasik** [comment]
Wouldn't it be a good idea to make the enzyme adapters peer dependencies and make the user include the corresponding one?

**Comment by ryan-roemer** [comment]
Making the enzyme adapters peer dependencies means that we'll have lots of peers that aren't required -- the vastly typical case I assume is a project uses exactly **one** adapter. And since the adapters are named with versions like `enzyme-adapter-react-16`, we can't do something like:

```js
"peerDependencies": {
  "enzyme-adapter-react": "^13.x || ^ 14.x || ^15.x"
}
```

**Comment by pascalduez** [comment]
`optionalDependencies` maybe? Even if it's not really optional... :\

**Comment by blainekasten** [comment]
`optionalDependencies` wouldn't actually _do_ anything for us. If you read the docs for it it still requires us to manually try and catch.

https://docs.npmjs.com/files/package.json#optionaldependencies

I'll try and get a PR up for this today and release a new version. It'll have to be a major since it requires work to update :(

**Comment by VishalGulati** [comment]
@blainekasten I am also facing the same issue. I use below config:

    "react": "^15.1.0",
    "enzyme": "^3.2.0",
    "react-addons-test-utils": "^15.3.1",
    "react-dom": "^15.1.0",

And I get below errors:

```
! peer dep missing: react@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-dom@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
```

I see in the node_modules/react/package.json that the version installed is 15.6.2 wereas I specifically installed 15.1 as prod dependency.

**Comment by blainekasten** [comment]
@VishalGulati I haven't fixed this yet

**Comment by VishalGulati** [comment]
@blainekasten Ok, thanks for your response. I will be following this chain for updates on this.

**Comment by blainekasten** [comment]
This is now fixed in 6.0.0
